### PR TITLE
Support gradient clipping by value in Fabric FSDP

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Gradient clipping by value is supported with FSDP out of the box, because it doesn't require any unsharding of gradients and can be done with local gradients. I am not sure why we had an explicit error for this, there must have been a mixup. Trainer already supports this. This PR also enables the test for it.

----
📚 Documentation preview 📚: https://pytorch-lightning--19236.org.readthedocs.build/en/19236/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by Borda — APPROVED** [review]
does it require any min version?
